src/common/string_utils.cpp

```diff
@@ -46,9 +46,14 @@ std::vector<std::string_view> StringUtils::smartSplit(std::string_view input, ch
     std::vector<std::string_view> result;
     auto currentItem = 0u;
     std::vector<char> stk;
+    bool insideSingleQuote = false;
     for (auto i = 0u; i < input.size(); i++) {
         char c = input[i];
-        if (c == splitChar && stk.size() == 0u) {
+
+        if (c == '\'' && (stk.size() == 0u || stk.back() != '\'')) {
```

Comment:
Do we have to care about double quote or other quotation characters?

Reply:
Double quotation has been handled in the previous implementation as an escapeChar. I think for the `smartsplit` , it only handles the value within one column and the quotation sign is guaranteed to be single?